custom/lib/Passage.php

```diff
@@ -107,6 +107,20 @@ public function getUser(string $user_id): UserInfo|Model401Error|Model404Error|M
         return $this->usersApi->getUser($this->appId, $user_id)['user'];
     }
 
+    /**
+     * Get a user by identifier
+     * @return UserInfo|Model401Error|Model404Error|Model500Error
+     */
+    public function getUserByIdentifier(string $identifier): UserInfo|Model401Error|Model404Error|Model500Error {
+        $users = $this->usersApi->listPaginatedUsers($this->appId, limit: 1, identifier:$identifier)['users']
+
+        if (count($users) == 0) {
+            // throw 404 exception
+        }
+
+        return $this->usersApi->getUser($this->appId, $users[0]['id'])['user'];
+    }
```

Comment:
tried my best but really struggling through the php. Really need help with catching error on 115 and throwing on 118. 

Here is the [go](https://github.com/passageidentity/passage-go/pull/53/files) 

and [python](https://github.com/passageidentity/passage-python/pull/59) example

Reply:
I don't know that we need to throw an error for 115. `$this->usersApi->listPaginatedUsers` throws all errors for us. 

118 We can utilize the generated [ApiException](https://github.com/passageidentity/passage-php/blob/main/generated/lib/ApiException.php) just like the generated files.

Reply:
Example, be sure to check this code:

```php
          throw new ApiException(
                    "[404] Could not find user with that identifier.",
                    404,
                    null,
                    null
          );
```

Reply:
Hmm thinking on this a bit more maybe it is a problem that we aren't catching and throwing very clear errors. We leave it to the generated errors but we probably shouldn't. Out of scope for this PR.